src/mlpack/methods/ann/loss_functions/huber_loss_impl.hpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file huber_loss_impl.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Implementation of the Huber Loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_HUBER_LOSS_IMPL_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_HUBER_LOSS_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "huber_loss.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+HuberLoss<InputDataType, OutputDataType>::HuberLoss(
+  const double delta,
+  const bool mean):
+  delta(delta),
+  mean(mean)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+double HuberLoss<InputDataType, OutputDataType>::Forward(
+    const InputType&& input, const TargetType&& target)
+{
+  double lossThis;
+  double totalLoss = 0;
+  double absError;
+
+  for (size_t i = 0; i < input.n_elem; ++i)
+  {
+      absError = std::abs(target[i] - input[i]);
```

Comment:
I guess you could also use `const double absError = std::abs(target[i] - input[i]);` to make clear the value isn't changed.